Add FW auto-trimming doc (#3146)
PX4/PX4-user_guide: en/config_fw/trimming_guide_fixedwing.md

1. Fly in stabilized mode at cruise speed and set the pitch setpoint offset (`FW_PSP_OFF`) to desired angle of attack.
   The required angle of attack at cruise speed corresponds to the pitch angle that the airplane needs to fly at in order to keep constant altitude during wing-leveled flight.
   If you are using an airspeed sensor, also set the correct cruise airspeed (`FW_AIRSPD_TRIM`).
1. Run [auto-trimming](#auto-trimming) during a flight to set `TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW`.

>    Alternatively, if you feel comfortable flying in manual mode, you can also set the trim values manually:
>    1. Disable auto-trimming using [FW_ATRIM_MODE](../advanced_config/parameter_reference.md#FW_ATRIM_MODE).
>    1. Fly in manual mode and trim your remote (with the trim switches).
>    1. Report the values to `TRIM_ROLL/PITCH/YAW` (and remove the trims from your transmitter) or update `TRIM_ROLL/PITCH/YAW` directly during flight via telemetry and QGC.

Reviewer: Just linking to manual mode to make it clear we mean "the" manual mode, not "a" manual mode. Also show value to set the ATRIM.

```suggestion
   Alternatively, if you feel comfortable flying in [Manual mode](../flight_modes_fw/manual.md), you can also set the trim values manually:
   1. Disable auto-trimming by setting [FW_ATRIM_MODE=0](../advanced_config/parameter_reference.md#FW_ATRIM_MODE).
   1. Fly in manual mode and trim your remote (with the trim switches).
   1. Report the values to `TRIM_ROLL/PITCH/YAW` (and remove the trims from your transmitter), or update `TRIM_ROLL/PITCH/YAW` directly during flight via telemetry and QGC.
```

### Auto-trimming

<Badge type="warning" text="main (PX4 v1.15)" />

> Auto-trimming is enabled by default.

Reviewer: ```suggestion
Auto-trimming estimates the "center trim" at cruise speed (usually), and is enabled by default.
```

> On first flight, auto-trimming estimates the current trim of the aircraft, and then updates the general trim parameters (`TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW`) with calibrated values after landing.

Reviewer: ```suggestion
On first flight (in "Calibration" mode), auto-trimming estimates the current trim of the aircraft, and then updates the general trim parameters (`TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW`) with calibrated values after landing.
```

The trim estimate is only updated when the aircraft flies with a bank angle of less than 10 degrees for more than 7 seconds.
So for the first calibration flight you should fly large circles or several phases of straight and level flight.

> On subsequent flights auto-trimming continuously estimates the trim and slowly updates the stored trim parameters.
> This reduces the constant load on the integrator of the angular rate control loop.

Reviewer: ```suggestion
On subsequent flights (in "Continuous" mode), auto-trimming continuously estimates the trim, reducing the constant load on the integrator of the angular rate control loop if the trim calibration was imperfect or if the trim has changed.
In this mode the stored trim parameters are updated gradually — the maximum change to the trim parameters following each flight is 0.05.
```

Auto-trimming is enabled using [FW_ATRIM_MODE](../advanced_config/parameter_reference.md#FW_ATRIM_MODE).

> This parameter is set to `(1) Calibration` (trim parameters are replaced by the trim estimate on landing) for the first flight and is then automatically changed to `(2) Continuous` mode (trim estimate is used to slowly update the existing trim parameters).
> 
> ## Advanced Trimming

Reviewer: ```suggestion
This parameter is set to `(1) Calibration` (trim parameters are replaced by the trim estimate on landing) for the first flight and is then automatically changed to `(2) Continuous` mode (trim estimate is used to slowly update the existing trim parameters).

The trims can be reset by setting the `TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW` back to `0` and `FW_ATRIM_MODE` to `1`.
This should not normally be necessary as the trims are continuously corrected. 

## Advanced Trimming
```

1. Fly in stabilized mode at cruise speed and set the pitch setpoint offset (`FW_PSP_OFF`) to desired angle of attack.
   The required angle of attack at cruise speed corresponds to the pitch angle that the airplane needs to fly at in order to keep constant altitude during wing-leveled flight.
   If you are using an airspeed sensor, also set the correct cruise airspeed (`FW_AIRSPD_TRIM`).
1. Run [auto-trimming](#auto-trimming) during a flight to set `TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW`.

>    You could also manually set the pitch trim (`TRIM_PITCH`) by looking at the actuator controls in the log file (upload it to [Flight Review](https://logs.px4.io) and check the _Actuator Controls_ plot for example).

Reviewer: FYI I split this to make the auto trimming the preferred.

Author: Good idea

   Set that value to the average offset of the pitch signal during wing-leveled flight.

Step 3 can be performed before step 2 if you don't want to have to look at the log, or if you feel comfortable flying in manual mode.

> You can then trim your remote (with the trim switches) and report the values to `TRIM_PITCH` (and remove the trims from your transmitter) or update `TRIM_PITCH` directly during flight via telemetry and QGC.

Reviewer: I am not sure about this stuff. It seems to only cover TRIM_PITCH? Is it still reasonable advice in the context of auto-trimming?

Author: I don't know why it was mentioning pitch only. I tried to rephrase it, lmk if that's clear enough

> You can then trim your remote (with the trim switches) and report the values to `TRIM_PITCH` (and remove the trims from your transmitter) or update `TRIM_PITCH` directly during flight via telemetry and QGC.

Author: ```suggestion
You can then trim your remote (with the trim switches) and report the values to `TRIM_ROLL/PITCH/YAW` (and remove the trims from your transmitter) or update `TRIM_ROLL/PITCH/YAW` directly during flight via telemetry and QGC.
```

> ### Auto-trimming

Reviewer: Can calibration fail or result in bad values? If so, what would the user see? How would they know that they need to do it again?

Author: Except if there is a bug, normally it can't give bad values. If the trim values are not perfect, it will anyway correct them in the next flight. A user will feel bad trim values if he flies manually (he'll need to counteract the effect).
One can always reset the trims by setting the `TRIM_XXX` back to 0 and `FW_ATRIM_MODE` to "Calibration.

> <Badge type="warning" text="main (PX4 v1.15)" />

Reviewer: FYI I added versioning tags (trialing this).

> Auto-trimming continuously estimates the current trim of the aircraft during flight, uses it to reduce the constant load on the integrator of the angular rate control loop, and adjusts general trim parameters (`TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW`) after landing.
> 
> Auto-trimming is enabled using [FW_ATRIM_MODE](../advanced_config/parameter_reference.md#FW_ATRIM_MODE).
> This parameter is set to `(1) Calibration` (trim parameters are replaced by the trim estimate on landing) for the first flight and is then automatically changed to `(2) Continuous` mode (trim estimate is used to slowly update the existing trim parameters).
> 
> The trim estimate is only updated when the aircraft constantly flies with a bank angle of less than 10 degrees for more than 7 seconds.
> During the first calibration flight you should fly large circles or several phases of straight and level flight to obtain best results.

Reviewer: Howdy @bresch 

Where you say 
1. "uses it to reduce the constant load on the integrator of the angular rate control loop". Why do we care?
2. Does the continuous update ever actually update the trim parameters, or does it just adjust the current values vs calibrated "on the fly"?

When I read instructions I want to know what I do, in order, whereas the way this is written it talks about what the feature does. You can infer one from the other, so it is a matter of preference. But would you be OK with something like this (adjusted with comments above).

```suggestion
Auto-trimming is enabled by default.
On first flight, auto-trimming estimates the current trim of the aircraft, and then updates the general trim parameters (`TRIM_ROLL`, `TRIM_PITCH` and `TRIM_YAW`) with calibrated values after landing.
The trim estimate is only updated when the aircraft flies with a bank angle of less than 10 degrees for more than 7 seconds.
So for the first calibration flight you should fly large circles or several phases of straight and level flight.

On subsequent flights auto-trimming continuously estimates the trim and slowly updates the stored trim parameters.
This reduces the constant load on the integrator of the angular rate control loop.

Auto-trimming is enabled using [FW_ATRIM_MODE](../advanced_config/parameter_reference.md#FW_ATRIM_MODE).
This parameter is set to `(1) Calibration` (trim parameters are replaced by the trim estimate on landing) for the first flight and is then automatically changed to `(2) Continuous` mode (trim estimate is used to slowly update the existing trim parameters).
```

> ## Advanced Trimming

Reviewer: My assumption is that advanced trimming is still needed?

Author: Yes, currently the auto-trim only estimates the "center trim" (usually at cruise speed)

Reviewer: Worth making that very clear - done in https://github.com/PX4/PX4-user_guide/pull/3146/files#r1550616689
